### PR TITLE
fix: stop order expiry

### LINKED ISSUE #8663: [Bug]: Stop orders expiring incorrectly
### Problem encountered

Stop orders are expiring before their expiry time.

### Observed behaviour

Stop orders are expiring before their intended expiry.

### Expected behaviour

Stop orders should expiry as per specified time on the stop order request.

### Steps to reproduce

System Test Name - test_stop_order_submit_expiry_bug
System Test Branch - feat_stop_order_bugs

```bash
Submitted Stop Order:

Submitting stop order with ref: test_stop_order_submit_expiry_bug[order_param0]-bdec448d {'stopOrdersSubmission': {'risesAbove': {'orderSubmission': {'marketId': '102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01', 'price': '130000000', 'size': '1000000', 'side': 'SIDE_SELL', 'timeInForce': 'TIME_IN_FORCE_FOK', 'type': 'TYPE_LIMIT', 'reference': 'test_stop_order_submit_expiry_bug[order_param0]-bdec448d', 'reduceOnly': True, 'expiresAt': '0', 'postOnly': False}, 'expiresAt': '1688403195372424000', 'expiryStrategy': 'EXPIRY_STRATEGY_SUBMIT', 'trailingPercentOffset': '0.99'}, 'fallsBelow': {'orderSubmission': {'marketId': '102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01', 'price': '130000000', 'size': '1000000', 'side': 'SIDE_SELL', 'timeInForce': 'TIME_IN_FORCE_FOK', 'type': 'TYPE_LIMIT', 'reference': 'test_stop_order_submit_expiry_bug[order_param0]-bdec448d', 'reduceOnly': True, 'expiresAt': '0', 'postOnly': False}, 'expiresAt': '1688403195372424000', 'expiryStrategy': 'EXPIRY_STRATEGY_SUBMIT', 'trailingPercentOffset': '0.99'}}}



Note the expiry on the stop order is 100 minutes from the current time.
```

### Software version

vega - develop

### Failing test

test_stop_order_submit_expiry_bug

### Jenkins run

https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/79974/

### Configuration used

_No response_

### Relevant log output

```shell
Jenkins Logs: 
https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/79974/consoleText

System Test Log:

2023-07-03 15:13:16.233:[INFO] [MainProcess] [MainThread] [stop_orders_test.py]:[test_stop_order_submit_expiry_bug]:[  42] Market data before placing stop order is {'markPrice': '97600000', 'bestBidPrice': '90000000', 'bestBidVolume': '500000', 'bestOfferPrice': '120000000', 'bestOfferVolume': '500000', 'bestStaticBidPrice': '90000000', 'bestStaticBidVolume': '500000', 'bestStaticOfferPrice': '120000000', 'bestStaticOfferVolume': '500000', 'midPrice': '105000000', 'staticMidPrice': '105000000', 'market': '102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01', 'timestamp': '1688397195372424000', 'openInterest': '1000000', 'indicativePrice': '0', 'marketTradingMode': 'TRADING_MODE_CONTINUOUS', 'targetStake': '2698152000000000000000', 'suppliedStake': '390500000000000000000000', 'marketValueProxy': '390500000000000000000000', 'liquidityProviderFeeShare': [{'party': '4a0de4f82518ebd83775edccdcb9356c8d74d2ae17ffdeb1459bcfacf4a48b7d', 'equityLikeShare': '1', 'averageEntryValuation': '390500000000000000000000', 'averageScore': '1', 'virtualStake': '390500000000000000000000'}], 'marketState': 'STATE_ACTIVE', 'nextMarkToMarket': '1688397195070617363', 'lastTradedPrice': '97600000', 'marketGrowth': '0', 'auctionEnd': '0', 'auctionStart': '0', 'indicativeVolume': '0', 'trigger': 'AUCTION_TRIGGER_UNSPECIFIED', 'extensionTrigger': 'AUCTION_TRIGGER_UNSPECIFIED', 'priceMonitoringBounds': []}
2023-07-03 15:13:16.250:[INFO] [MainProcess] [MainThread] [  ordersMod.py]:[stop_order_submit]:[ 548] Submitting stop order with ref: test_stop_order_submit_expiry_bug[order_param0]-bdec448d {'stopOrdersSubmission': {'risesAbove': {'orderSubmission': {'marketId': '102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01', 'price': '130000000', 'size': '1000000', 'side': 'SIDE_SELL', 'timeInForce': 'TIME_IN_FORCE_FOK', 'type': 'TYPE_LIMIT', 'reference': 'test_stop_order_submit_expiry_bug[order_param0]-bdec448d', 'reduceOnly': True, 'expiresAt': '0', 'postOnly': False}, 'expiresAt': '1688403195372424000', 'expiryStrategy': 'EXPIRY_STRATEGY_SUBMIT', 'trailingPercentOffset': '0.99'}, 'fallsBelow': {'orderSubmission': {'marketId': '102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01', 'price': '130000000', 'size': '1000000', 'side': 'SIDE_SELL', 'timeInForce': 'TIME_IN_FORCE_FOK', 'type': 'TYPE_LIMIT', 'reference': 'test_stop_order_submit_expiry_bug[order_param0]-bdec448d', 'reduceOnly': True, 'expiresAt': '0', 'postOnly': False}, 'expiresAt': '1688403195372424000', 'expiryStrategy': 'EXPIRY_STRATEGY_SUBMIT', 'trailingPercentOffset': '0.99'}}}
2023-07-03 15:13:16.250:[INFO] [MainProcess] [MainThread] [   clientv2.py]:[    __init__]:[  38] instantiating WalletClientV2: http://127.0.0.1:1789, /jenkins/workspace/common/system-tests-wrapper/system-tests/vegacapsule/api-token-passphrase-file, /jenkins/workspace/common/system-tests-wrapper/networkdata/testnet/wallet
2023-07-03 15:13:16.251:[INFO] [MainProcess] [MainThread] [   clientv2.py]:[       login]:[ 174] logging in to: 6AB69C1417
2023-07-03 15:13:16.259:[INFO] [MainProcess] [MainThread] [   clientv2.py]:[   signtx_v2]:[ 370] calling jsonrpc client.send_transaction for wallet: 6AB69C1417, pubkey: 8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499: req: {'stopOrdersSubmission': {'risesAbove': {'orderSubmission': {'marketId': '102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01', 'price': '130000000', 'size': '1000000', 'side': 'SIDE_SELL', 'timeInForce': 'TIME_IN_FORCE_FOK', 'type': 'TYPE_LIMIT', 'reference': 'test_stop_order_submit_expiry_bug[order_param0]-bdec448d', 'reduceOnly': True, 'expiresAt': '0', 'postOnly': False}, 'expiresAt': '1688403195372424000', 'expiryStrategy': 'EXPIRY_STRATEGY_SUBMIT', 'trailingPercentOffset': '0.99'}, 'fallsBelow': {'orderSubmission': {'marketId': '102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01', 'price': '130000000', 'size': '1000000', 'side': 'SIDE_SELL', 'timeInForce': 'TIME_IN_FORCE_FOK', 'type': 'TYPE_LIMIT', 'reference': 'test_stop_order_submit_expiry_bug[order_param0]-bdec448d', 'reduceOnly': True, 'expiresAt': '0', 'postOnly': False}, 'expiresAt': '1688403195372424000', 'expiryStrategy': 'EXPIRY_STRATEGY_SUBMIT', 'trailingPercentOffset': '0.99'}}, 'propagate': True, 'pubKey': '8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499'}
2023-07-03 15:13:16.260:[INFO] [MainProcess] [MainThread] [   clientv2.py]:[     _header]:[  53] using token: NZ3cSlAUZItuLOJsQlqJSqH2bzzhKiIltDuZxdSUh6j05WxB9Iw8tfSv4wUbtzB1
2023-07-03 15:13:16.268:[INFO] [MainProcess] [MainThread] [   clientv2.py]:[   signtx_v2]:[ 395] response: {'jsonrpc': '2.0', 'result': {'receivedAt': '2023-07-03T15:13:16.265294943Z', 'sentAt': '2023-07-03T15:13:16.266654359Z', 'transactionHash': 'B7EEEFC1073AA0B12720DED131C0D84409EDECFD891A0457AA8D4A754A9DB0F7', 'transaction': {'input_data': 'CLuwzKTonNeAnwEQ7gHCP9gCCqkBCpMBCkAxMDJiNGNhYzdlOTkwMTU1YWQyZjIyN2M5NzRmN2QxOGY3M2UxYWFmZGJkY2FmMDBlZTdmZjk1NjZkMTMzYTAxEgkxMzAwMDAwMDAYwIQ9IAIoBDgBQjh0ZXN0X3N0b3Bfb3JkZXJfc3VibWl0X2V4cGlyeV9idWdbb3JkZXJfcGFyYW0wXS1iZGVjNDQ4ZFgBEMCWu86ruJq3FxgCqgYEMC45ORKpAQqTAQpAMTAyYjRjYWM3ZTk5MDE1NWFkMmYyMjdjOTc0ZjdkMThmNzNlMWFhZmRiZGNhZjAwZWU3ZmY5NTY2ZDEzM2EwMRIJMTMwMDAwMDAwGMCEPSACKAQ4AUI4dGVzdF9zdG9wX29yZGVyX3N1Ym1pdF9leHBpcnlfYnVnW29yZGVyX3BhcmFtMF0tYmRlYzQ0OGRYARDAlrvOq7iatxcYAqoGBDAuOTk=', 'signature': {'value': 'cd60181fba1ed198220fba858c16f548ef80903ada34631a357a3bf43ab6e7818bd0da48ac35312429fcf157dbe40f21f6f963462cd408958640eca48fdb6b0d', 'algo': 'vega/ed25519', 'version': 1}, 'From': {'PubKey': '8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499'}, 'version': 3, 'pow': {'tid': '30caa250bb91195c49bfbba07deee1bc26a979e41c2294cc5c6eb2c039a144df', 'nonce': 8}}}, 'id': 'request'}
2023-07-03 15:13:16.269:[INFO] [MainProcess] [MainThread] [ walletsMod.py]:[wallet_sign_submit_txn_v2]:[ 630] TRX request b'{"jsonrpc": "2.0", "method": "client.send_transaction", "params": {"publicKey": "8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499", "sendingMode": "TYPE_SYNC", "transaction": {"stopOrdersSubmission": {"risesAbove": {"orderSubmission": {"marketId": "102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01", "price": "130000000", "size": "1000000", "side": "SIDE_SELL", "timeInForce": "TIME_IN_FORCE_FOK", "type": "TYPE_LIMIT", "reference": "test_stop_order_submit_expiry_bug[order_param0]-bdec448d", "reduceOnly": true, "expiresAt": "0", "postOnly": false}, "expiresAt": "1688403195372424000", "expiryStrategy": "EXPIRY_STRATEGY_SUBMIT", "trailingPercentOffset": "0.99"}, "fallsBelow": {"orderSubmission": {"marketId": "102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01", "price": "130000000", "size": "1000000", "side": "SIDE_SELL", "timeInForce": "TIME_IN_FORCE_FOK", "type": "TYPE_LIMIT", "reference": "test_stop_order_submit_expiry_bug[order_param0]-bdec448d", "reduceOnly": true, "expiresAt": "0", "postOnly": false}, "expiresAt": "1688403195372424000", "expiryStrategy": "EXPIRY_STRATEGY_SUBMIT", "trailingPercentOffset": "0.99"}}}}, "id": "request"}'
2023-07-03 15:13:16.269:[INFO] [MainProcess] [MainThread] [ walletsMod.py]:[wallet_sign_submit_txn_v2]:[ 631] TRX submitted in epoch 25 block 238 status 200 OK signed by trader/pubkey 6AB69C1417/8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499
2023-07-03 15:13:16.269:[INFO] [MainProcess] [MainThread] [ walletsMod.py]:[wallet_sign_submit_txn_v2]:[ 634] TRX response {"receivedAt": "2023-07-03T15:13:16.265294943Z", "sentAt": "2023-07-03T15:13:16.266654359Z", "transactionHash": "B7EEEFC1073AA0B12720DED131C0D84409EDECFD891A0457AA8D4A754A9DB0F7", "transaction": {"input_data": "CLuwzKTonNeAnwEQ7gHCP9gCCqkBCpMBCkAxMDJiNGNhYzdlOTkwMTU1YWQyZjIyN2M5NzRmN2QxOGY3M2UxYWFmZGJkY2FmMDBlZTdmZjk1NjZkMTMzYTAxEgkxMzAwMDAwMDAYwIQ9IAIoBDgBQjh0ZXN0X3N0b3Bfb3JkZXJfc3VibWl0X2V4cGlyeV9idWdbb3JkZXJfcGFyYW0wXS1iZGVjNDQ4ZFgBEMCWu86ruJq3FxgCqgYEMC45ORKpAQqTAQpAMTAyYjRjYWM3ZTk5MDE1NWFkMmYyMjdjOTc0ZjdkMThmNzNlMWFhZmRiZGNhZjAwZWU3ZmY5NTY2ZDEzM2EwMRIJMTMwMDAwMDAwGMCEPSACKAQ4AUI4dGVzdF9zdG9wX29yZGVyX3N1Ym1pdF9leHBpcnlfYnVnW29yZGVyX3BhcmFtMF0tYmRlYzQ0OGRYARDAlrvOq7iatxcYAqoGBDAuOTk=", "signature": {"value": "cd60181fba1ed198220fba858c16f548ef80903ada34631a357a3bf43ab6e7818bd0da48ac35312429fcf157dbe40f21f6f963462cd408958640eca48fdb6b0d", "algo": "vega/ed25519", "version": 1}, "From": {"PubKey": "8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499"}, "version": 3, "pow": {"tid": "30caa250bb91195c49bfbba07deee1bc26a979e41c2294cc5c6eb2c039a144df", "nonce": 8}}, "txHash": "B7EEEFC1073AA0B12720DED131C0D84409EDECFD891A0457AA8D4A754A9DB0F7", "tx": {"input_data": "CLuwzKTonNeAnwEQ7gHCP9gCCqkBCpMBCkAxMDJiNGNhYzdlOTkwMTU1YWQyZjIyN2M5NzRmN2QxOGY3M2UxYWFmZGJkY2FmMDBlZTdmZjk1NjZkMTMzYTAxEgkxMzAwMDAwMDAYwIQ9IAIoBDgBQjh0ZXN0X3N0b3Bfb3JkZXJfc3VibWl0X2V4cGlyeV9idWdbb3JkZXJfcGFyYW0wXS1iZGVjNDQ4ZFgBEMCWu86ruJq3FxgCqgYEMC45ORKpAQqTAQpAMTAyYjRjYWM3ZTk5MDE1NWFkMmYyMjdjOTc0ZjdkMThmNzNlMWFhZmRiZGNhZjAwZWU3ZmY5NTY2ZDEzM2EwMRIJMTMwMDAwMDAwGMCEPSACKAQ4AUI4dGVzdF9zdG9wX29yZGVyX3N1Ym1pdF9leHBpcnlfYnVnW29yZGVyX3BhcmFtMF0tYmRlYzQ0OGRYARDAlrvOq7iatxcYAqoGBDAuOTk=", "signature": {"value": "cd60181fba1ed198220fba858c16f548ef80903ada34631a357a3bf43ab6e7818bd0da48ac35312429fcf157dbe40f21f6f963462cd408958640eca48fdb6b0d", "algo": "vega/ed25519", "version": 1}, "From": {"PubKey": "8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499"}, "version": 3, "pow": {"tid": "30caa250bb91195c49bfbba07deee1bc26a979e41c2294cc5c6eb2c039a144df", "nonce": 8}}}


Wallet Log:

Trying to send the transaction: pub_key:"8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499"  stop_orders_submission:{rises_above:{order_submission:{market_id:"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01"  price:"130000000"  size:1000000  side:SIDE_SELL  time_in_force:TIME_IN_FORCE_FOK  type:TYPE_LIMIT  reference:"test_stop_order_submit_expiry_bug[order_param0]-bdec448d"  reduce_only:true}  expires_at:1688403195372424000  expiry_strategy:EXPIRY_STRATEGY_SUBMIT  trailing_percent_offset:"0.99"}  falls_below:{order_submission:{market_id:"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01"  price:"130000000"  size:1000000  side:SIDE_SELL  time_in_force:TIME_IN_FORCE_FOK  type:TYPE_LIMIT  reference:"test_stop_order_submit_expiry_bug[order_param0]-bdec448d"  reduce_only:true}  expires_at:1688403195372424000  expiry_strategy:EXPIRY_STRATEGY_SUBMIT  trailing_percent_offset:"0.99"}}

Stop Order Events:

{"id":"240-5","block":"D391BC9BB0859412D0A3E2E468D0EE0F4297C3D9E211AA96CB054BF607C3E54A","type":"BUS_EVENT_TYPE_STOP_ORDER","stopOrder":{"submission":{"marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","price":"130000000","size":"1000000","side":"SIDE_SELL","timeInForce":"TIME_IN_FORCE_FOK","type":"TYPE_LIMIT","reference":"test_stop_order_submit_expiry_bug[order_param0]-bdec448d","reduceOnly":true},"stopOrder":{"id":"2a82c1828baf77ab88a5bb2d0fae56050c4270a83bebfb129921e141361831a3","ocoLinkId":"ff63de4064b4ebff4522237ed78693339b40ca3c185a0c7d90a91f3cd4fa31ef","expiresAt":"1688403195372424000","expiryStrategy":"EXPIRY_STRATEGY_SUBMIT","triggerDirection":"TRIGGER_DIRECTION_FALLS_BELOW","status":"STATUS_PENDING","createdAt":"1688397196","updatedAt":"1688397196","partyId":"8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499","marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","trailingPercentOffset":"0.99"}},"version":1,"chainId":"testnet-001","txHash":"B7EEEFC1073AA0B12720DED131C0D84409EDECFD891A0457AA8D4A754A9DB0F7"}
{"id":"240-6","block":"D391BC9BB0859412D0A3E2E468D0EE0F4297C3D9E211AA96CB054BF607C3E54A","type":"BUS_EVENT_TYPE_STOP_ORDER","stopOrder":{"submission":{"marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","price":"130000000","size":"1000000","side":"SIDE_SELL","timeInForce":"TIME_IN_FORCE_FOK","type":"TYPE_LIMIT","reference":"test_stop_order_submit_expiry_bug[order_param0]-bdec448d","reduceOnly":true},"stopOrder":{"id":"ff63de4064b4ebff4522237ed78693339b40ca3c185a0c7d90a91f3cd4fa31ef","ocoLinkId":"2a82c1828baf77ab88a5bb2d0fae56050c4270a83bebfb129921e141361831a3","expiresAt":"1688403195372424000","expiryStrategy":"EXPIRY_STRATEGY_SUBMIT","triggerDirection":"TRIGGER_DIRECTION_RISES_ABOVE","status":"STATUS_PENDING","createdAt":"1688397196","updatedAt":"1688397196","partyId":"8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499","marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","trailingPercentOffset":"0.99"}},"version":1,"chainId":"testnet-001","txHash":"B7EEEFC1073AA0B12720DED131C0D84409EDECFD891A0457AA8D4A754A9DB0F7"}
{"id":"240-7","block":"D391BC9BB0859412D0A3E2E468D0EE0F4297C3D9E211AA96CB054BF607C3E54A","type":"BUS_EVENT_TYPE_STOP_ORDER","stopOrder":{"submission":{"marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","price":"130000000","size":"1000000","side":"SIDE_SELL","timeInForce":"TIME_IN_FORCE_FOK","type":"TYPE_LIMIT","reference":"test_stop_order_submit_expiry_bug[order_param0]-bdec448d","reduceOnly":true},"stopOrder":{"id":"2a82c1828baf77ab88a5bb2d0fae56050c4270a83bebfb129921e141361831a3","ocoLinkId":"ff63de4064b4ebff4522237ed78693339b40ca3c185a0c7d90a91f3cd4fa31ef","expiresAt":"1688403195372424000","expiryStrategy":"EXPIRY_STRATEGY_SUBMIT","triggerDirection":"TRIGGER_DIRECTION_FALLS_BELOW","status":"STATUS_PENDING","createdAt":"1688397196","updatedAt":"1688397196","partyId":"8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499","marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","trailingPercentOffset":"0.99"}},"version":1,"chainId":"testnet-001","txHash":"B7EEEFC1073AA0B12720DED131C0D84409EDECFD891A0457AA8D4A754A9DB0F7"}
{"id":"240-8","block":"D391BC9BB0859412D0A3E2E468D0EE0F4297C3D9E211AA96CB054BF607C3E54A","type":"BUS_EVENT_TYPE_STOP_ORDER","stopOrder":{"submission":{"marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","price":"130000000","size":"1000000","side":"SIDE_SELL","timeInForce":"TIME_IN_FORCE_FOK","type":"TYPE_LIMIT","reference":"test_stop_order_submit_expiry_bug[order_param0]-bdec448d","reduceOnly":true},"stopOrder":{"id":"ff63de4064b4ebff4522237ed78693339b40ca3c185a0c7d90a91f3cd4fa31ef","ocoLinkId":"2a82c1828baf77ab88a5bb2d0fae56050c4270a83bebfb129921e141361831a3","expiresAt":"1688403195372424000","expiryStrategy":"EXPIRY_STRATEGY_SUBMIT","triggerDirection":"TRIGGER_DIRECTION_RISES_ABOVE","status":"STATUS_PENDING","createdAt":"1688397196","updatedAt":"1688397196","partyId":"8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499","marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","trailingPercentOffset":"0.99"}},"version":1,"chainId":"testnet-001","txHash":"B7EEEFC1073AA0B12720DED131C0D84409EDECFD891A0457AA8D4A754A9DB0F7"}
{"id":"241-3","block":"4D469042D159A13D997F1FD7C1ECC635664464F9692D8D3330D9B3C0E59FDA61","type":"BUS_EVENT_TYPE_STOP_ORDER","stopOrder":{"submission":{"marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","price":"130000000","size":"1000000","side":"SIDE_SELL","timeInForce":"TIME_IN_FORCE_FOK","type":"TYPE_LIMIT","reference":"test_stop_order_submit_expiry_bug[order_param0]-bdec448d","reduceOnly":true},"stopOrder":{"id":"2a82c1828baf77ab88a5bb2d0fae56050c4270a83bebfb129921e141361831a3","ocoLinkId":"ff63de4064b4ebff4522237ed78693339b40ca3c185a0c7d90a91f3cd4fa31ef","expiresAt":"1688403195372424000","expiryStrategy":"EXPIRY_STRATEGY_SUBMIT","triggerDirection":"TRIGGER_DIRECTION_FALLS_BELOW","status":"STATUS_EXPIRED","createdAt":"1688397196","updatedAt":"1688397196","partyId":"8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499","marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","trailingPercentOffset":"0.99"}},"version":1,"chainId":"testnet-001","txHash":"4D469042D159A13D997F1FD7C1ECC635664464F9692D8D3330D9B3C0E59FDA61"}
{"id":"241-4","block":"4D469042D159A13D997F1FD7C1ECC635664464F9692D8D3330D9B3C0E59FDA61","type":"BUS_EVENT_TYPE_STOP_ORDER","stopOrder":{"submission":{"marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","price":"130000000","size":"1000000","side":"SIDE_SELL","timeInForce":"TIME_IN_FORCE_FOK","type":"TYPE_LIMIT","reference":"test_stop_order_submit_expiry_bug[order_param0]-bdec448d","reduceOnly":true},"stopOrder":{"id":"ff63de4064b4ebff4522237ed78693339b40ca3c185a0c7d90a91f3cd4fa31ef","ocoLinkId":"2a82c1828baf77ab88a5bb2d0fae56050c4270a83bebfb129921e141361831a3","expiresAt":"1688403195372424000","expiryStrategy":"EXPIRY_STRATEGY_SUBMIT","triggerDirection":"TRIGGER_DIRECTION_RISES_ABOVE","status":"STATUS_EXPIRED","createdAt":"1688397196","updatedAt":"1688397196","partyId":"8c71b918ecebca8e72b3a9267494fb93f109b37283685362c5f071fb7d03e499","marketId":"102b4cac7e990155ad2f227c974f7d18f73e1aafdbdcaf00ee7ff9566d133a01","trailingPercentOffset":"0.99"}},"version":1,"chainId":"testnet-001","txHash":"4D469042D159A13D997F1FD7C1ECC635664464F9692D8D3330D9B3C0E59FDA61"}
```

### PR DESCRIPTION
**Pull request by jeremyletang**
close #8663

### CODE REVIEW

File: core/execution/engine.go [hunk 1]
```diff
@@ -659,7 +659,7 @@ func (e *Engine) SubmitStopOrders(
 	submission *types.StopOrdersSubmission,
 	party string,
 	idgen common.IDGenerator,
-	fallsBelowID *string,
+	fallsBellowID *string,
```
Comment (ze97286) [inline core/execution/engine.go:662]: below

File: core/execution/future/market.go [hunk 2]
```diff
@@ -1353,13 +1353,13 @@ func (m *Market) SubmitStopOrdersWithIDGeneratorAndOrderIDs(
 	submission *types.StopOrdersSubmission,
 	party string,
 	idgen common.IDGenerator,
-	fallsBelowID, risesAboveID *string,
+	fallsBellowID, risesAboveID *string,
```
Comment (ze97286) [inline core/execution/future/market.go:1356]: Bellow => Below everywhere in this file

File: core/execution/stoporders/priced_stop_orders.go [hunk 3]
```diff
@@ -34,15 +34,15 @@ type PricedStopOrders struct {
 	// help finding them back in the trees.
 	orders map[string]orderAtPriceStat
 
-	fallsBelow *btree.BTreeG[*ordersAtPrice]
-	risesAbove *btree.BTreeG[*ordersAtPrice]
+	fallsBellow *btree.BTreeG[*ordersAtPrice]
```
Comment (ze97286) [inline core/execution/stoporders/priced_stop_orders.go:37]: same here about Bellow

File: core/execution/stoporders/stop_orders_snapshot_test.go [hunk 4]
```diff
@@ -53,4 +55,12 @@ func TestStopOrdersSnapshot(t *testing.T) {
 	buf2, err := proto.Marshal(serialized2)
 	assert.NoError(t, err)
 	assert.Equal(t, hex.EncodeToString(buf), hex.EncodeToString(buf2))
+
+	m := jsonpb.Marshaler{}
```
Comment (ze97286) [inline core/execution/stoporders/stop_orders_snapshot_test.go:59]: still need this?

Comment (jeremyletang) [reply core/execution/stoporders/stop_orders_snapshot_test.go:59]: nope

**Review by ze97286 — DISMISSED** [review]

**Review by ValentinTrinque — DISMISSED** [review]

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**